distributed/protocol/serialize.py

```diff
@@ -473,6 +474,8 @@ def replace_inner(x):
 
 def serialize_bytelist(x, **kwargs):
     header, frames = serialize(x, **kwargs)
+    if "lengths" not in header:
+        header["lengths"] = tuple(map(nbytes, frames))
     frames = sum(map(frame_split_size, frames), [])
```

Comment:
Note that the frames are already be split here. This is due to constraints caused by compression (which happens below).